### ISSUE TITLE
Enable MAGMA TPL with new CMake 

### ISSUE THREAD

**Issue by vqd8a** [kokkos/kokkos-kernels#713]
@jjwilke @e10harvey 
I am trying building Kokkos Kernels within Trilinos with MAGMA TPL enabled. 
I used:
```
export MAGMA_ROOT=/g/g92/dang15/Packages/install-magma 
-D KokkosKernels_ENABLE_TPL_MAGMA:BOOL=ON
```
as said in the [build instructions](https://github.com/kokkos/kokkos-kernels/blob/master/BUILD.md) 
but it reports that it cannot find the MAGMA headers (```magma_v2.h```, ```magma_lapack.h```)

I had to do these to make it compiled
```
-D CMAKE_CXX_FLAGS="-O3 -g -I${MAGMA_ROOT}/include" \
-D Trilinos_EXTRA_LINK_FLAGS:STRING="-L${MAGMA_ROOT}/lib -lmagma"
```

Do you have any idea what went wrong? Is it because we don't have something like 
```FindTPLMAGMA.cmake``` in the ```cmake/Modules``` directory?

**Comment by jjwilke** [comment]
I started this in PR #714. That hopefully is close enough to get you going.

**Comment by vqd8a** [comment]
Thanks, @jjwilke .

**Comment by e10harvey** [comment]
Addressed by #982.
